jsonschema/src/compilation/options.rs

```diff
@@ -584,6 +590,50 @@ impl CompilationOptions {
     pub(crate) const fn are_unknown_formats_ignored(&self) -> bool {
         self.ignore_unknown_formats
     }
+
+    /// Register a custom keyword definition.
+    ///
+    /// Examples
+    ///
+    /// ```rust
+    /// # use jsonschema::{CustomKeywordDefinition, ErrorIterator, JSONSchema, paths::JSONPointer};
+    /// # use serde_json::{json, Value};
+    /// # use std::sync::Arc;
+    ///
+    /// fn validate(
+    ///     instance: &Value,
+    ///     instance_path: JSONPointer,
+    ///     schema: Arc<Value>,
+    ///     schema_path: JSONPointer,
+    /// ) -> ErrorIterator {
+    ///     // ... validate instance ...
+    ///     Box::new(None.into_iter())
+    /// }
+    /// fn is_valid(instance: &Value, schema: &Value) -> bool {
+    ///     // ... determine if instance is valid ...
+    ///     return true;
+    /// }
+    /// let definition = CustomKeywordDefinition::Validator { validate, is_valid };
+    /// assert!(JSONSchema::options()
+    ///     .with_custom_keyword("my-type", definition)
+    ///     .compile(&json!({ "my-type": "my-schema"}))
+    ///     .expect("A valid schema")
+    ///     .is_valid(&json!({ "a": "b"})));
+    /// ```
+    pub fn with_custom_keyword<T>(mut self, keyword: T, definition: CustomKeywordDefinition) -> Self
```

Comment:
I believe this does not need to consume `self`, so maybe make it take in and return `&mut self` just like `with_format()` above? This would allow it to chain method calls more easily.